engine/src/flutter/lib/web_ui/test/canvaskit/surface_test.dart

```diff
@@ -213,21 +213,22 @@ void testMain() {
 
         // Emulate WebGL context loss.
         final DomOffscreenCanvas canvas = surface.debugGetOffscreenCanvas()!;
-        final Object ctx = canvas.getContext('webgl2')!;
-        final Object loseContextExtension = js_util.callMethod(ctx, 'getExtension', <String>[
-          'WEBGL_lose_context',
-        ]);
-        js_util.callMethod<void>(loseContextExtension, 'loseContext', const <void>[]);
+        final JSObject ctx = canvas.getContext('webgl2')! as JSObject;
+        final JSObject loseContextExtension = ctx.callMethod<JSObject>(
+          'getExtension'.toJS,
+          'WEBGL_lose_context'.toJS,
+        );
```

Comment:
Instead of using unsafe interop, you could go one step further and add the [`getExtension`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/getExtension) method in `dom.dart` here:
https://github.com/flutter/flutter/blob/9ceee4496d95b28156b7d1603e6550957629bfff/engine/src/flutter/lib/web_ui/lib/src/engine/dom.dart#L829-L837